runtime/compiler/net/MessageBuffer.hpp

```diff
@@ -204,18 +206,31 @@ class MessageBuffer
 
    uint32_t getCapacity() const { return _capacity; }
 
+   // Try to free the custom persistent allocator for message buffers. This method does nothing
+   // if the JVM is not in client mode, or if there is at least one active message buffer.
+   static void tryFreePersistentAllocator();
 
 private:
    static const size_t INITIAL_BUFFER_SIZE = 32768; // Initial buffer size is 32K
    uint32_t offset(char *addr) const { return addr - _storage; }
-   char *allocateMemory(uint32_t capacity) { return static_cast<char *>(_allocator.allocate(capacity)); }
-   void freeMemory(char *storage) { _allocator.deallocate(storage); }
+   char *allocateMemory(uint32_t capacity) { return static_cast<char *>(_allocator->allocate(capacity)); }
+   void freeMemory(char *storage) { _allocator->deallocate(storage); }
    uint32_t computeRequiredCapacity(uint32_t requiredSize);
 
+   static TR::Monitor *getTotalBuffersMonitor()
+      {
+      if (!_totalBuffersMonitor)
```

Comment:
There is a race here: two threads executing this routine at the same time could allocate a monitor each.
Maybe we should create this monitor ahead of time, during bootstrap.

Reply:
That's probably clearer. I assumed that there would only be one thread trying to initiate contact with a server at the start of a JVM instance.